src/plugins/data_source/server/client/data_source_client.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Client } from '@opensearch-project/opensearch';
+import {
+  Logger,
+  OpenSearchClient,
+  SavedObject,
+  SavedObjectsClientContract,
+  SavedObjectsErrorHelpers,
+} from '../../../../../src/core/server';
+import { CREDENTIAL_SAVED_OBJECT_TYPE, DATA_SOURCE_SAVED_OBJECT_TYPE } from '../../common';
+import {
+  CredentialMaterials,
+  CredentialSavedObjectAttributes,
+} from '../../common/credentials/types';
+import { DataSourceAttributes } from '../../common/data_sources';
+import { DataSourcePluginConfigType } from '../../config';
+import { DataSourceServiceSetup } from '../data_source_service';
+
+/**
+ * Represents an OpenSearch cluster API client created by the platform.
+ * It allows to call API on behalf of the user(credential) associated to "data source"
+ *
+ * @public
+ **/
+export interface IDataSourceClient {
+  /**
+   * Creates a {@link OpenSearchClient } bound to given data source
+   */
+  asDataSource: (dataSourceId: string) => Promise<OpenSearchClient>;
+}
+
+interface DataSourceClientCtorParams {
```

Comment:
What does Ctor mean? could we use full name

Reply:
removed in the new version